### PR TITLE
Change caching scripts to a declarative pipeline

### PR DESCRIPTION
**Pull request by coopie**
Previously, processing the data for experiements involved running 2 large scripts. Now, training is done using generators which rely on a pipeline that is exposed to them.

Note: TTV stands for test, train, validation split of data. Whenever a class is instantiated with a ttv, it is a dict in the form:
```
{
  "test": {
      userid: [uri]
      ...
  }
  "train": {
   ...
  }

  ...
}
```

### CODE REVIEW

**Comment by coopie** [comment]
Travis will always crash at the moment due to dependency issues. If CI is really needed, I can run the tests in a docker container in the future.

File: speech_ml/util.py [hunk 1]
```diff
@@ -45,15 +45,20 @@ def mkdir_p(path):
 
 
 def ttv_yaml_to_dict(path):
+    """The naming of this is bad."""
```
Comment (cmiller01) [inline speech_ml/util.py]: Maybe you want to fix this now 😉

File: test/data_sources_test.py [hunk 2]
```diff
@@ -72,22 +69,196 @@ def test_spectrogram_data_source(self):
...
+    def test_cached_ttv_array_like_data_source(self):
+        dummy_data_source = DummyDataSource()
+        subject_info_dir = os.path.join('test', 'dummy_data', 'metadata')
+        ttv = yaml_to_dict(os.path.join(subject_info_dir, 'dummy_ttv.yaml'))
+
+        array_ds = CachedTTVArrayLikeDataSource(dummy_data_source, ttv, data_name='dummy', cache_name='test')
+
+        self.assertEqual(len(array_ds), 3)
+
+        # all_values = np.fromiter((x for x in array_ds[:]), dtype='int16')
```
Comment (cmiller01) [inline test/data_sources_test.py]: Do you want to remove this?

**Comment by cmiller01** [comment]
Couple notes on cleanup. Otherwise LGTM

**Comment by coopie** [comment]
Actually, changed my mind, feel free to merge now without squashing. I thought there were more junk commits.

**Comment by HongdaMao** [comment]
the code didn't pass through travis. Can you take a look?

**Comment by coopie** [comment]
Ok, tests are passing on CI. Used a fix from here http://conda.pydata.org/docs/travis.html as previously everything broke because travis had to manually compile the native libraries used in scientific python packages.

**Pull request merged**